language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -273,34 +292,45 @@ computeCek con t@(Var _ varName)   = do
 returnCek
     :: (GShow uni, GEq uni, DefaultUni <: uni, Closed uni, uni `Everywhere` ExMemoryUsage)
     => Context uni -> WithMemory Value uni -> CekM uni (Plain Term uni)
--- Instantiate all the free variable of the resulting term in case there are any.
+-- Instantiate all the free variables of the resulting term in case there are any.
 returnCek [] res = getVarEnv <&> \varEnv -> void $ dischargeVarEnv varEnv res
-returnCek (FrameTyInstArg ty : con) fun = instantiateEvaluate con ty fun
-returnCek (FrameApplyArg argVarEnv arg : con) fun = do
+returnCek (FrameTyInstArg _ : ctx) fun =
+    case fun of
```

Comment:
(I don't mind inlining the match)

Reply:
I think I originally had separate matches but then changed it because there are quite a lot of cases in `returnCek` and I thought it'd make it easier to see what's going on if there was exactly one top-level match for each kind of frame.